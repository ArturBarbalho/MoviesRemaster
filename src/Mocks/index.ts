import { BannerProps } from "../typesProps";

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }