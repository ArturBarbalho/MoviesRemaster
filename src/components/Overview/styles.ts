import styled,{css} from 'styled-components'
type props = {
    size:'small'|'medium'|'large'
}

const Modifier = {
    small:()=> css`
    max-width:900px;
    margin-top:30px;
    
    `,
    medium:()=> css`
    max-width:1000px;
    margin-top:60px;
    @media(max-width:900px){
        font-size:15px;
        margin-top:10px;
}
    `,
    large:()=> css`
    max-width:1500px;
    width:100vw;
    margin-left:20px;
    ` 
}


export const Wrapper = styled.p<props>`
${({size})=>css`
color:white;
font-size:larger;
letter-spacing: 1px;
text-shadow:1px 0px 1px black;
${Modifier[size]()}
`}
`