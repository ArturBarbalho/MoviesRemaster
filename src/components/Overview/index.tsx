import * as S from './styles'
type props = {
  overview:string,
  size:'small'|'medium'|'large'
}
const Overview = ({overview,size}:props) => (
  <S.Wrapper size={size} >{overview}</S.Wrapper>
)

export default Overview