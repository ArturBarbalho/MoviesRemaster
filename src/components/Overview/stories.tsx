import { Story, Meta } from '@storybook/react/types-6-0'
import Overview from '.'
import { BannerMock } from '../../Mocks'
export default {
  title: 'Overview',
  component: Overview
} as Meta

export const Default: Story = () => <Overview size='small' overview={BannerMock.overview} />