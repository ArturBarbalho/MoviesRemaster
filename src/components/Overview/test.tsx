import { render, screen } from '@testing-library/react'

import Overview from '.'
import { BannerMock } from '../../Mocks'

describe('<Overview />', () => {
  it('should render the overview text and the small size', () => {
    const { container } = render(<Overview size='small' overview={BannerMock.overview} />)

    expect(screen.getByText( BannerMock.overview )).toBeInTheDocument()
    expect(screen.getByText( BannerMock.overview )).toHaveStyle(
      'max-width:900px;'
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the medium size', () => {
    const { container } = render(<Overview size='medium' overview={BannerMock.overview} />)

    expect(screen.getByText( BannerMock.overview )).toHaveStyle(
      'max-width:1000px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the large size', () => {
    const { container } = render(<Overview size='large' overview={BannerMock.overview} />)

    expect(screen.getByText( BannerMock.overview )).toHaveStyle(
      'max-width:1500px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})