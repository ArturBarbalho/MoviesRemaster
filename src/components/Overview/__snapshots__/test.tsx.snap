// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> should render the large size 1`] = `
<p
  class="styles__Wrapper-sc-mhbed0-0 gIpRWZ"
>
  Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
</p>
`;

exports[`<Overview /> should render the medium size 1`] = `
<p
  class="styles__Wrapper-sc-mhbed0-0 FTjwm"
>
  Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
</p>
`;

exports[`<Overview /> should render the overview text and the small size 1`] = `
<p
  class="styles__Wrapper-sc-mhbed0-0 fSwUgG"
>
  Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
</p>
`;
